alacritty/src/ipc.rs

```diff
@@ -55,9 +86,13 @@ pub fn spawn_ipc_socket(options: &Options, event_proxy: EventLoopProxy<Event>) -
 }
 
 /// Send a message to the active Alacritty socket.
-pub fn send_message(socket: Option<PathBuf>, message: &[u8]) -> IoResult<()> {
-    let socket = find_socket(socket)?;
-    socket.send(message)?;
+pub fn send_message(socket: Option<PathBuf>, message: IPCMessage) -> IoResult<()> {
+    let mut socket = find_socket(socket)?;
+
+    let mut message = serde_json::to_string(&message)?;
```

Comment:
I just wanted to point out that back when I looked at this, I was considering a binary exchange format. I think in theory that would be pretty nice because the data sent over the wire would be much more efficiently encoded.

But from what I've seen that would require pulling another big dependency just for doing this, so I think just using what we have already is probably smart.

Reply:
I was also thinking about using protobuf, msgpack or some other fancy thing, but then I see serde and unix sockets in rust stdlib, so I decided to pick whatever I have, since the bandwidth is low anyway and json is fine here...